[BUG] fixed flow rand init as global tensor bug

grybd opened the pull request:
这个PR修复 module forward 中构造 rand tensor 会报错的bug，相关背景记录在issue : #7340

strint commented:
这个我上午review下。

你再另外做个工作，就是检查下scouce op，就是没有输入的op，看看是否还有别的没有支持。

关于哪些是souce op，你可以请教下 @BBuf

grybd commented:
> 这个我上午review下。
> 
> 你再另外做个工作，就是检查下scouce op，就是没有输入的op，看看是否还有别的没有支持。
> 
> 关于哪些是souce op，你可以请教下 @BBuf

好的

strint commented on python/oneflow/test/modules/test_rand.py:
```diff
@@ -106,5 +106,44 @@ def test_cases(test_case):
+        ]
+        arg_dict["sbp"] = [(flow.sbp.broadcast,), (flow.sbp.split(0),)]
+        for arg in GenArgList(arg_dict):
+            arg[0](test_case, *arg[1:])
```
单测可以正常通过不

grybd replied:
是可以通过的。

strint commented on oneflow/user/ops/distributions/uniform_op.cpp:
```diff
@@ -30,7 +31,22 @@ namespace oneflow {
+  }
+
+  *ctx->OutputShape("out", 0) = Shape(dim_vec);
+  return Maybe<void>::Ok();
```
另外其它还有哪些source op，可以在这个pr下面comment一下，作为任务列表

grybd replied:
好的。

grybd commented:
- [x] OneFlow_UniformOp  支持 S、添加单测
- [] OneFlow_ArangeOp 支持 S、添加单测
- [] OneFlow_CoinFlipOp 支持 S、添加单测
- [] OneFlow_EyeOp  支持 S、添加单测
- [] OneFlow_RandpermOp 支持 S、添加单测
- [] OneFlow_RecvOp 支持 S、添加单测
- []  OneFlow_NormalOp 支持 S、添加单测
- [] OneFlow_FlushSummaryWriterOp 支持 S、添加单测
- [] OneFlow_GenTensorBufferOp 支持 S、添加单测
- [] OneFlow_TestRandomSourceOp 支持 S、添加单测
- [] OneFlow_TestSourceOp  支持 S、添加单测
- [] OneFlow_TestSourceMultiGpuFixedOutNumOp 支持 S、添加单测

grybd commented:
问题复现
- [x] OneFlow_UniformOp  
```python
import oneflow as flow

class TestGraph(flow.nn.Graph):
    def __init__(self):
        super().__init__()

    def build(self, x):
        shape = (x.shape[0], 1, 1)
        output = x * flow.rand(*shape, dtype=x.dtype, placement=x.placement, sbp=x.sbp)
        # 改成下面的代码则不会报错
        # output = x * flow.randint(0, 1, shape, placement=x.placement, sbp=x.sbp)
        return output

tensor = flow.ones(32, 100, 100, placement=flow.env.all_device_placement("cuda"), sbp=flow.sbp.split(0))
test = TestGraph()
out = test(tensor)
```
```shell
命令： python3 -m oneflow.distributed.launch --nproc_per_node 2 --master_port 12345 test.py
```
报错信息：
```shell
F0111 12:10:01.930294 28125 exec_graph.cpp:122]                                                                                                
[82/99]  File "../oneflow/core/graph/exec_graph.cpp", line 122, in InferBlobDescs                                                                                
CheckPhysicalBlobDesc( *op(), op()->output_bns(), std ... nd_sbp_signature, parallel_ctx, GetBlobDesc4BnInOp)

  File "../oneflow/core/graph/exec_graph.cpp", line 97, in CheckPhysicalBlobDesc
    CheckPhysicalBlobDesc(*::oneflow::private_details:: ... op_parallel_desc, parallel_ctx, *physical_blob_desc)

  File "../oneflow/core/graph/exec_graph.cpp", line 78, in CheckPhysicalBlobDesc

    Check failed: (physical.shape()) == (*::oneflow::private_details::RemoveRValConst(({ 
auto&& _just_value_to_check_ = GetPhysicalShape(logical.shape(), nd_sbp, parallel_desc,
 *parallel_ctx); if (!::oneflow::private_details::JustIsOk(_just_value_to_check_)) { return 
::oneflow::private_details::JustErrorAddStackFrame( 
::oneflow::private_details::JustGetError(_just_value_to_check_), 
"../oneflow/core/graph/exec_graph.cpp", 78, __FUNCTION__, 
"GetPhysicalShape(logical.shape(), nd_sbp, parallel_desc, *parallel_ctx)"); } 
std::forward<decltype(_just_value_to_check_)>(_just_value_to_check_); 
})).Data_YouAreNotAllowedToCallThisFuncOutsideThisFile()) ((32,1,1) vs (16,1,1))

*** Check failure stack trace: ***
    @     0x7f3765588f8d  google::LogMessage::Fail()
    @     0x7f376558a84d  google::LogMessage::SendToLog()
    @     0x7f3765588a4d  google::LogMessage::Flush()
    @     0x7f376558c4a9  google::LogMessageFatal::~LogMessageFatal()                                                                                     
    @     0x7f37567ab71b  oneflow::ExecNode::InferBlobDescs()                                                                                             
    @     0x7f37567b0103  oneflow::Graph<>::TopoForEachNodeWithErrorCaptured()
    @     0x7f37567b0d46  oneflow::Graph<>::TopoForEachNode()
    @     0x7f375682b66f  oneflow::NormalForwardCompTaskNode::BuildExecGphAndRegst()
    @     0x7f37567ff1a9  oneflow::Graph<>::TopoForEachNodeWithErrorCaptured()
    @     0x7f37567ff98b  oneflow::Graph<>::TopoForEachNode()
    @     0x7f375687f856  oneflow::Compiler::Compile()
    @     0x7f37563e37e4  oneflow::NNGraph::CompileAndInitRuntime()
```
- [ ] OneFlow_RandpermOp 
复现问题的代码：
```python
import oneflow as flow

class TestGraph(flow.nn.Graph):
    def __init__(self):
        super().__init__()

    def build(self, x):
        output = x * flow.randperm(10, dtype=x.dtype, placement=x.placement, sbp=x.sbp)
        # 改成下面的代码则不会报错
        # output = x * flow.randint(0, 1, shape, placement=x.placement, sbp=x.sbp)
        return output

tensor = flow.randn(10, placement=flow.env.all_device_placement("cuda"), sbp=flow.sbp.split(0))
test = TestGraph()
out = test(tensor)
```
```shell
命令：python3 -m oneflow.distributed.launch --nproc_per_node 2 --master_port 12345 /home/fengdaochao/fdz/test_randperm.py
```
```shell
报错信息：
F20220126 15:08:41.819885 2327043 exec_graph.cpp:119] 
  File "/home/fengdaochao/fdz/oneflow/oneflow/core/graph/exec_graph.cpp", line 119, in InferBlobDescs
    CheckPhysicalBlobDesc( *op(), op()->output_bns(), std ... nd_sbp_signature, parallel_ctx, GetBlobDesc4BnInOp)
  File "/home/fengdaochao/fdz/oneflow/oneflow/core/graph/exec_graph.cpp", line 95, in CheckPhysicalBlobDesc
    CheckPhysicalBlobDesc(*::oneflow::private_details:: ... op_parallel_desc, parallel_ctx, *physical_blob_desc)
  File "/home/fengdaochao/fdz/oneflow/oneflow/core/graph/exec_graph.cpp", line 77, in CheckPhysicalBlobDesc
    Check failed: (physical.shape()) == (*::oneflow::private_details::RemoveRValConst(({ auto&& _just_value_to_check_ = GetPhysicalShape(logical.shape(), nd_sbp, parallel_desc, *parallel_ctx); if (!::oneflow::private_details::JustIsOk(_just_value_to_check_)) { return ::oneflow::private_details::JustErrorAddStackFrame( ::oneflow::private_details::JustGetError(_just_value_to_check_), "/home/fengdaochao/fdz/oneflow/oneflow/core/graph/exec_graph.cpp", 77, __FUNCTION__, "GetPhysicalShape(logical.shape(), nd_sbp, parallel_desc, *parallel_ctx)"); } std::forward<decltype(_just_value_to_check_)>(_just_value_to_check_); })).Data_YouAreNotAllowedToCallThisFuncOutsideThisFile()) ((10,) vs (5,))
*** Check failure stack trace: ***
    @     0x7fb2116de400  (unknown)
    @     0x7fb2116e14c1  (unknown)
    @     0x7fb2116ddedc  (unknown)
    @     0x7fb2116e1bbf  (unknown)
    @     0x7fb20c8211e4  oneflow::ExecNode::InferBlobDescs()
    @     0x7fb20c8232a7  _ZNSt17_Function_handlerIFN7oneflow5MaybeIvvEEPNS0_8ExecNodeEEZZNKS0_5GraphIS3_NS0_8ExecEdgeEE15TopoForEachNodeERKNSt7__cxx114listIS4_SaIS4_EEERKSt8functionIFvS4_RKSF_IFvS4_EEEESN_SJ_ENKUlPKcE_clESP_EUlS4_E_E9_M_invokeERKSt9_Any_dataOS4_
    @     0x7fb20c825467  oneflow::Graph<>::TopoForEachNodeWithErrorCaptured()
    @     0x7fb20c825f7d  oneflow::Graph<>::TopoForEachNode()
    @     0x7fb20c89f6bb  oneflow::NormalForwardCompTaskNode::BuildExecGphAndRegst()
    @     0x7fb20c878157  _ZNSt17_Function_handlerIFN7oneflow5MaybeIvvEEPNS0_8TaskNodeEEZZNKS0_5GraphIS3_NS0_8TaskEdgeEE15TopoForEachNodeERKNSt7__cxx114listIS4_SaIS4_EEERKSt8functionIFvS4_RKSF_IFvS4_EEEESN_SJ_ENKUlPKcE_clESP_EUlS4_E_E9_M_invokeERKSt9_Any_dataOS4_
    @     0x7fb20c87dc26  oneflow::Graph<>::TopoForEachNodeWithErrorCaptured()
    @     0x7fb20c87e6c6  oneflow::Graph<>::TopoForEachNode()
    @     0x7fb20c8eef47  oneflow::Compiler::Compile()
    @     0x7fb20c3da266  oneflow::NNGraph::CompileAndInitRuntime()
    @     0x7fb20a73abb8  (unknown)
    @     0x7fb20aadde78  (unknown)
    @           0x5f5e79  PyCFunction_Call
    @           0x5f6a46  _PyObject_MakeTpCall
    @           0x50b4a7  (unknown)
    @           0x5703e6  _PyEval_EvalFrameDefault
    @           0x5696da  _PyEval_EvalCodeWithName
    @           0x5f6403  _PyFunction_Vectorcall
    @           0x50ad5c  (unknown)
    @           0x5f55f2  PyObject_Call
    @           0x56cbfb  _PyEval_EvalFrameDefault
    @           0x5696da  _PyEval_EvalCodeWithName
    @           0x5f6403  _PyFunction_Vectorcall
    @           0x59cffd  (unknown)
    @           0x5f6a46  _PyObject_MakeTpCall
    @           0x570612  _PyEval_EvalFrameDefault
    @           0x5696da  _PyEval_EvalCodeWithName
    @           0x68db17  PyEval_EvalCode
F20220126 15:08:42.210513 2327044 rpc_client.cpp:40] Check failed: stub->CallMethod<ctrl_method>(&client_ctx, request_, &response_).error_code() == grpc::StatusCode::OK (14 vs. 0) 
*** Check failure stack trace: ***
    @     0x7fd22e3ec400  (unknown)
    @     0x7fd22e3ef4c1  (unknown)
    @     0x7fd22e3ebedc  (unknown)
    @     0x7fd22e3efbbf  (unknown)
    @     0x7fd227bdd9de  (unknown)
    @     0x7fd227bdf1c2  oneflow::RpcClient::PullKV()
    @     0x7fd227bdf338  oneflow::RpcClient::PullKV()
    @     0x7fd2290e873b  oneflow::NNGraph::CompileAndInitRuntime()
    @     0x7fd227448bb8  (unknown)
    @     0x7fd2277ebe78  (unknown)
    @           0x5f5e79  PyCFunction_Call
    @           0x5f6a46  _PyObject_MakeTpCall
    @           0x50b4a7  (unknown)
    @           0x5703e6  _PyEval_EvalFrameDefault
    @           0x5696da  _PyEval_EvalCodeWithName
    @           0x5f6403  _PyFunction_Vectorcall
    @           0x50ad5c  (unknown)
    @           0x5f55f2  PyObject_Call
    @           0x56cbfb  _PyEval_EvalFrameDefault
    @           0x5696da  _PyEval_EvalCodeWithName
    @           0x5f6403  _PyFunction_Vectorcall
    @           0x59cffd  (unknown)
    @           0x5f6a46  _PyObject_MakeTpCall
    @           0x570612  _PyEval_EvalFrameDefault
    @           0x5696da  _PyEval_EvalCodeWithName
    @           0x68db17  PyEval_EvalCode
    @           0x67eeb1  (unknown)
    @           0x67ef2f  (unknown)
    @           0x67efd1  (unknown)
    @           0x67f377  PyRun_SimpleFileExFlags
    @           0x6b7902  Py_RunMain
    @           0x6b7c8d  Py_BytesMain
Killing subprocess 2327043
Killing subprocess 2327044
```
- [ ] OneFlow_EyeOp  支持 S、添加单测
```python

- [ ] OneFlow_ArangeOp 支持 S、添加单测
- [ ] OneFlow_CoinFlipOp 支持 S、添加单测
- [ ] OneFlow_RecvOp 支持 S、添加单测
- [ ]  OneFlow_NormalOp 支持 S、添加单测
- [ ] OneFlow_FlushSummaryWriterOp 支持 S、添加单测
- [ ] OneFlow_GenTensorBufferOp 支持 S、添加单测
- [ ] OneFlow_TestRandomSourceOp 支持 S、添加单测
- [ ] OneFlow_TestSourceOp  支持 S、添加单测
- [ ] OneFlow_TestSourceMultiGpuFixedOutNumOp 支持 S、添加单测

grybd commented:
目前讲randperm_op.cpp 下的
```cpp
/*static*/ Maybe<void> RandpermOp::InferPhysicalTensorDesc(user_op::InferContext* ctx) {
  return InferLogicalTensorDesc(ctx);
}
```
改为
```cpp
/*static*/ Maybe<void> RandpermOp::InferPhysicalTensorDesc(user_op::InferContext* ctx) {
  const Shape& shape = ctx->Attr<Shape>("shape");
  DimVector dim_vec{shape.dim_vec()};

  const cfg::SbpParallel& out_sbp_para = ctx->SbpParallel4ArgNameAndIndex("out", 0);
  if (out_sbp_para.has_split_parallel()) {
    const int64_t& parallel_num = ctx->parallel_ctx().parallel_num();
    if (parallel_num > 1) {
      const int64_t& split_axis = out_sbp_para.split_parallel().axis();
      CHECK_LT_OR_RETURN(split_axis, dim_vec.size());
      BalancedSplitter bs(shape.At(split_axis), parallel_num);
      dim_vec[split_axis] = bs.At(ctx->parallel_ctx().parallel_id()).size();
    }
  }

  *ctx->OutputShape("out", 0) = Shape(dim_vec);
  return Maybe<void>::Ok();
}
/*static*/ Maybe<void> RandpermOp::InferDataType(user_op::InferContext* ctx) {
  *ctx->OutputDType("out", 0) = DataType::kInt32;
  return Maybe<void>::Ok();
}
```
还是会报错：
```shell

sEnvironment has been initialized, this env init will do nothing.
F20220126 14:04:22.900836 2845208 user_op_conf.cpp:86] Check failed: attr.get() != nullptr attr_name: shape
*** Check failure stack trace: ***
    @     0x7f50a6f5a700  (unknown)
    @     0x7f50a6f5d7c1  (unknown)
    @     0x7f50a6f5a1dc  (unknown)
    @     0x7f50a6f5debf  (unknown)
    @     0x7f50a1d44711  oneflow::user_op::UserOpConfWrapper::Attr4Name()
    @     0x7f50a45c67ac  oneflow::RandpermOp::InferPhysicalTensorDesc()
    @     0x7f50a03df616  std::_Function_handler<>::_M_invoke()
    @     0x7f50a30afe07  oneflow::UserOp::InferOutBlobDescs()
    @     0x7f50a30696f3  oneflow::Operator::InferOutBlobDescsIf()
    @     0x7f50a30712c8  oneflow::Operator::InferBlobDescsIf()
    @     0x7f50a2097ba4  oneflow::ExecNode::InferBlobDescs()
    @     0x7f50a209a387  _ZNSt17_Function_handlerIFN7oneflow5MaybeIvvEEPNS0_8ExecNodeEEZZNKS0_5GraphIS3_NS0_8ExecEdgeEE15TopoForEachNodeERKNSt7__cxx114listIS4_SaIS4_EEERKSt8functionIFvS4_RKSF_IFvS4_EEEESN_SJ_ENKUlPKcE_clESP_EUlS4_E_E9_M_invokeERKSt9_Any_dataOS4_
    @     0x7f50a209c547  oneflow::Graph<>::TopoForEachNodeWithErrorCaptured()
    @     0x7f50a209d05d  oneflow::Graph<>::TopoForEachNode()
    @     0x7f50a211679b  oneflow::NormalForwardCompTaskNode::BuildExecGphAndRegst()
    @     0x7f50a20ef237  _ZNSt17_Function_handlerIFN7oneflow5MaybeIvvEEPNS0_8TaskNodeEEZZNKS0_5GraphIS3_NS0_8TaskEdgeEE15TopoForEachNodeERKNSt7__cxx114listIS4_SaIS4_EEERKSt8functionIFvS4_RKSF_IFvS4_EEEESN_SJ_ENKUlPKcE_clESP_EUlS4_E_E9_M_invokeERKSt9_Any_dataOS4_
    @     0x7f50a20f4d06  oneflow::Graph<>::TopoForEachNodeWithErrorCaptured()
    @     0x7f50a20f57a6  oneflow::Graph<>::TopoForEachNode()
    @     0x7f50a2165e57  oneflow::Compiler::Compile()
    @     0x7f50a1c53546  oneflow::NNGraph::CompileAndInitRuntime()
    @     0x7f509ffb94d8  (unknown)
    @     0x7f50a035ad38  (unknown)
    @           0x5f5b29  PyCFunction_Call
    @           0x5f66f6  _PyObject_MakeTpCall
    @           0x50ad17  (unknown)
    @           0x570296  _PyEval_EvalFrameDefault
    @           0x56951a  _PyEval_EvalCodeWithName
    @           0x5f60b3  _PyFunction_Vectorcall
    @           0x50a5cc  (unknown)
    @           0x5f54e7  PyObject_Call
    @           0x56ca9e  _PyEval_EvalFrameDefault
    @           0x56951a  _PyEval_EvalCodeWithName
Aborted (core dumped)
```
复现代码：
```python
import oneflow as flow

class TestGraph(flow.nn.Graph):
    def __init__(self):
        super().__init__()

    def build(self, x):
        output = x * flow.randperm(10, dtype=x.dtype, placement=x.placement, sbp=x.sbp)
        # 改成下面的代码则不会报错
        # output = x * flow.randint(0, 1, shape, placement=x.placement, sbp=x.sbp)
        return output

tensor = flow.randn(10, placement=flow.env.all_device_placement("cuda"), sbp=flow.sbp.split(0))
test = TestGraph()
out = test(tensor)
```
```shell
python3 -m oneflow.distributed.launch --nproc_per_node 2 --master_port 12345 test.py
```

grybd commented:
#### source op 支持 S，找不到用户接口的op
- [ ]  OneFlow_CoinFlipOp 支持 S、添加单测
找不到用户接口
- [ ]  OneFlow_RecvOp 支持 S、添加单测
找不到用户接口
- [ ]  OneFlow_NormalOp 支持 S、添加单测
找不到用户接口，还没和torch对齐
```python
import oneflow as flow
import torch

a = torch.normal(mean=0.5, std=torch.arange(1., 6.))
print("torch a: ",a)
b = flow.normal(mean=0.5, std=torch.arange(1., 6.))
print("flow b: ",b)
```
输出结果：
```shell
torch a:  tensor([-0.4274,  0.0511, -4.2247,  0.5314,  1.5141])
Traceback (most recent call last):
  File "/home/fengdaochao/fdz/test_normal.py", line 6, in <module>
    b = flow.normal(mean=0.5, std=torch.arange(1., 6.))
AttributeError: module 'oneflow' has no attribute 'normal'
```
- [ ] OneFlow_FlushSummaryWriterOp 支持 S、添加单测
该op 没有输入和输出，没有用户接口
- [ ] OneFlow_GenTensorBufferOp 支持 S、添加单测
还不知道如何写测试代码见接口信息：
```python
def gen_tensor_buffer(
    shape: Sequence[int],
    shape_list: Sequence[Sequence[int]],
    value_list: Sequence[float],
    data_type: Optional[flow.dtype] = flow.float32,
    dynamic_out: Optional[bool] = False,
):
    return flow._C.gen_tensor_buffer(
        shape, shape_list, value_list, data_type, dynamic_out
    )
```
- [ ] OneFlow_TestRandomSourceOp 支持 S、添加单测
找不到用户接口
- [ ] OneFlow_TestSourceOp  支持 S、添加单测
找不到用户接口
- [ ] OneFlow_TestSourceMultiGpuFixedOutNumOp 支持 S、添加单测
找不到用户接口，但shenghang好像支持S了

grybd commented:
####  randperm 和 arange op 支持 S 
由于 randperm 和 arange op的输出是 tensor的shape=[N],所以只能支持 S0 
```python
from pyrfc3339 import generate
import oneflow as flow

placement = flow.placement("cuda",{0:[0,1]})
sbp = flow.sbp.split(0)
x = flow.arange(1,3,0.5,placement=placement, sbp=sbp)

print("x.shape: ",x.shape)
print("x is consisten tensor: ",x.is_consistent)
print("consistent x: ",x)
print("local x: ",x.to_local())
print("local x shape: ",x.to_local().shape)
print("----------------------------------------------------------")

generator = flow.Generator()
generator.manual_seed(0)

y = flow.randperm(10,generator=generator,placement=placement,sbp=flow.sbp.split(0))
print("y shape: ",y.shape)
print("y is consisten tensor: ",y.is_consistent)
print("y consistent tensor: ",y)
print("y local_tensor: ",y.to_local())
print("local y shape: ",y.to_local().shape)
```
Terminal 0输入命令：```shell 
export MASTER_ADDR=127.0.0.1 MASTER_PORT=17789 WORLD_SIZE=2 RANK=0 LOCAL_RANK=0
python3 test.py
```
Terminal 1输入命令：```shell 
export MASTER_ADDR=127.0.0.1 MASTER_PORT=17789 WORLD_SIZE=2 RANK=1 LOCAL_RANK=1
python3 test.py
```
Terminal 0  输出：
```shell
x.shape:  oneflow.Size([4])
x is consisten tensor:  True
consistent x:  tensor([1.0000, 1.5000, 1.0000, 1.5000],
       placement=oneflow.placement(device_type="cuda", device_ids={0 : [0, 1]}, hierarchy=(2,)),
       sbp=(oneflow.sbp.split(axis=0),), dtype=oneflow.float32)
local x:  tensor([1.0000, 1.5000], device='cuda:1', dtype=oneflow.float32)
local x shape:  oneflow.Size([2])
----------------------------------------------------------
y shape:  oneflow.Size([10])
y is consisten tensor:  True
y consistent tensor:  tensor([7, 0, 9, 2, 5, 7, 0, 9, 2, 5],
       placement=oneflow.placement(device_type="cuda", device_ids={0 : [0, 1]}, hierarchy=(2,)),
       sbp=(oneflow.sbp.split(axis=0),), dtype=oneflow.int64)
y local_tensor:  tensor([7, 0, 9, 2, 5], device='cuda:1', dtype=oneflow.int64)
local y shape:  oneflow.Size([5])
```
Terminal 1 输出：
```shell
x.shape:  oneflow.Size([4])
x is consisten tensor:  True
consistent x:  tensor([1.0000, 1.5000, 1.0000, 1.5000],
       placement=oneflow.placement(device_type="cuda", device_ids={0 : [0, 1]}, hierarchy=(2,)),
       sbp=(oneflow.sbp.split(axis=0),), dtype=oneflow.float32)
local x:  tensor([1.0000, 1.5000], device='cuda:0', dtype=oneflow.float32)
local x shape:  oneflow.Size([2])
----------------------------------------------------------
y shape:  oneflow.Size([10])
y is consisten tensor:  True
y consistent tensor:  tensor([7, 0, 9, 2, 5, 7, 0, 9, 2, 5],
       placement=oneflow.placement(device_type="cuda", device_ids={0 : [0, 1]}, hierarchy=(2,)),
       sbp=(oneflow.sbp.split(axis=0),), dtype=oneflow.int64)
y local_tensor:  tensor([7, 0, 9, 2, 5], device='cuda:0', dtype=oneflow.int64)
local y shape:  oneflow.Size([5])
```
randperm 和 arange op的输出是 tensor的shape=[N],所以S1会报错：
```python
from pyrfc3339 import generate
import oneflow as flow

placement = flow.placement("cuda",{0:[0,1]})
sbp = flow.sbp.split(1)
x = flow.arange(1,3,0.5,placement=placement, sbp=sbp)
```
报错信息：
```shell
Check failed: index < this->NumAxes() (1 vs. 1)  Shape: (4,) visit index: 1 > num_axes: 1
*** Check failure stack trace: ***
    @     0x7f18ce58f700  (unknown)
    @     0x7f18ce5927c1  (unknown)
    @     0x7f18ce58f1dc  (unknown)
    @     0x7f18ce592ebf  (unknown)
    @     0x7f18c7d563f0  oneflow::Shape::At()
    @     0x7f18ca6a1971  oneflow::GetPhysicalShape()
    @     0x7f18c935e03c  oneflow::one::EagerConsistentTensorImpl::New()
    @     0x7f18c92bf0f0  (unknown)
    @     0x7f18c92c13e9  (unknown)
    @     0x7f18c92bbf2e  oneflow::one::EagerConsistentInterpreter::ApplyImpl()
    @     0x7f18c92ec3aa  oneflow::one::EagerInterpreter::Apply()
    @     0x7f18c92ed169  oneflow::one::AutogradInterpreter::Apply()
    @     0x7f18c92f0984  oneflow::one::OpInterpUtil::Dispatch()
    @     0x7f18c92f1850  oneflow::one::OpInterpUtil::Dispatch<>()
    @     0x7f18c92f1b65  oneflow::one::OpInterpUtil::Dispatch<>()
    @     0x7f18c9509c5c  oneflow::one::functional::impl::ConsistentArangeFunctor::operator()()
```
